app/scripts/directives/bindService.js

```diff
@@ -151,89 +177,33 @@ function BindService($filter,
         }
         sortServiceInstances();
       });
+      ctrl.appToBind = ctrl.target;
     }
     // TODO: handle not having any service instances when binding app to service
-    ctrl.gotoStep(ctrl.steps[0]);
   };
 
-  var humanizeKind = $filter('humanizeKind');
-  ctrl.groupByKind = function(object) {
-    return humanizeKind(object.kind);
-  };
-
-  var gotoStepID = function(id) {
-    var step = _.find(ctrl.steps, { id: id });
-    ctrl.gotoStep(step);
-  };
-
-  ctrl.gotoStep = function(step) {
-    _.each(ctrl.steps, function(st) {
-      st.selected = false;
-    });
-    if (ctrl.currentStep) {
-      ctrl.currentStep.visited = true;
+  ctrl.$onDestroy = function() {
+    if (validityWatcher) {
+      validityWatcher();
+      validityWatcher = undefined;
     }
-    ctrl.currentStep = step;
-    ctrl.currentStep.selected = true;
-  };
-
-  ctrl.stepClick = function(step) {
-    // Prevent returning to previous steps if the order is complete.
-    if (ctrl.wizardComplete) {
-      return;
-    }
-
-    if (!step.visited) {
-      return;
-    }
-
-    ctrl.gotoStep(step);
-  };
-
-  var generateName = $filter('generateName');
-  var makeBinding = function() {
-    var serviceInstanceName = ctrl.serviceToBind;
-    // TODO - would be better if generateName could take in an optional maxlength
-    var truncatedSvcInstanceName = _.trunc(serviceInstanceName, DNS1123_SUBDOMAIN_VALIDATION.maxlength - 6);
-    ctrl.generatedSecretName = generateName(truncatedSvcInstanceName + '-');
-    var binding = {
-      kind: 'Binding',
-      apiVersion: 'servicecatalog.k8s.io/v1alpha1',
-      metadata: {
-        generateName: serviceInstanceName + '-'
-      },
-      spec: {
-        instanceRef: {
-          name: serviceInstanceName
-        },
-        secretName: ctrl.generatedSecretName
-      }
-    };
-
-    return binding;
   };
 
   ctrl.bindService = function() {
     var svcToBind = ctrl.target.kind === 'Instance' ? ctrl.target : ctrl.serviceInstances[ctrl.serviceToBind];
+    var appToBind = ctrl.target.kind !== 'Instance' ? ctrl.target : ctrl.appToBind;
+
     var context = {
       namespace: _.get(svcToBind, 'metadata.namespace')
     };
-    DataService.create({
-      group: 'servicecatalog.k8s.io',
-      resource: 'bindings'
-    }, null, makeBinding(), context).then(function(binding) {
+
+    BindingService.bindService(context, _.get(svcToBind, 'metadata.name'), _.get(appToBind, 'metadata.name')).then(function(binding){
       ctrl.binding = binding;
+      ctrl.error = null;
 
-      DataService.watchObject({
-        group: 'servicecatalog.k8s.io',
-        resource: 'bindings'
-      }, _.get(ctrl.binding, 'metadata.name'), context, function(binding) {
+      DataService.watchObject(BindingService.bindingResource, _.get(ctrl.binding, 'metadata.name'), context, function(binding) {
```

Comment:
Not a new problem, but don't we need to be removing the watch when on scope destroy?